t/t0024-before-murder_once.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+. ./test-lib.sh
+t_plan 5 "before murder is called only once"
+
+t_begin "setup and start" && {
+	unicorn_setup
+	cat >> $unicorn_config <<EOF
+timeout 5
+after_fork { |s,w| File.open('$fifo','w') { |f| f.write '.' } }
+class Unicorn::HttpServer
+  def kill_worker(signal, wpid)
+    @iteration ||= 0
+    \$stderr.puts "kill_worker #{signal} #{wpid} #{@iteration}"
+    if @iteration > 0 then
+      Process.kill(signal, wpid)
+    end
+    @iteration = @iteration + 1
+  rescue Errno::ESRCH
+    @before_murder_called.delete(wpid)
+    worker = @workers.delete(wpid) and worker.close rescue nil
+  end
+end
```

Comment:
Might be nicer to use `alias_method` here, so:
```ruby
class Unicorn::HttpServer
  alias_method :kill_worker_always, :kill_worker
  def kill_worker(signal, wpid)
    @iteration ||= 0
    \$stderr.puts "kill_worker #{signal} #{wpid} #{@iteration}"
    kill_worker_always(signal, wpid) if @iteration > 0
    @iteration += 1
  end
end
```
This ensures we don't need to update the test if the implementation of `kill_worker` changes.